Infrastructure/Modules/CDNHelpers.psm1

```diff
@@ -0,0 +1,204 @@
+        }
+        else {
+            Write-Log -LogLevel Information -Message "Blob copy not required as OriginType set to either 'Cloud Service', 'Web App' or 'Custom Origin'"
+        }
+    }
+    catch {
+        throw "Failed to copy content to blob and set MIME settings: $_"
+    }
+}
+
+function Enable-CORS {
+    <#
+.SYNOPSIS
+Set CORS settings on blob storage
+
+.DESCRIPTION
+Cross-Origin Resource Sharing (CORS) is a mechanism that uses additional HTTP headers to tell a browser
+to let a web application running at one origin (domain) have permission to access selected resources from a server at a different origin.
+Courtesy: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+More Information: https://docs.microsoft.com/en-us/powershell/module/azure.storage/set-azurestoragecorsrule?view=azurermps-6.12.0
+
+.PARAMETER StorageAccountName
+The StorageAccountName to apply the CORS settings
+
+.PARAMETER SaSToken
+The SaS Token to access the blob storage container
+
+.PARAMETER EnableCORS
+The CORS Rules to be applied
+
+.EXAMPLE
+
+$DeploymentParameters = @ {
+    StorageAccountName = "mystorageaccountname"
+    SaSToken = "MySecureSaStokenString"
+    $EnableCORS = "Yes"
+}
+Enable-CORS @DeploymentParameters
+
+#>
+
+    Param(
+        [Parameter(Mandatory = $true)]
+        [String]$StorageAccountName,
+        [Parameter(Mandatory = $true)]
+        [String]$SaSToken,
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Yes", "No")]
+        [string]$EnableCORS
+    )
+    try {
+        # ---- Default CORS Settings
+        $CORSRules = (@{
+                AllowedHeaders  = @("*");
+                AllowedOrigins  = @("*");
+                MaxAgeInSeconds = 3600;
+                AllowedMethods  = @("Get")
+            })
+        # ---- Set CORS Rules
+        if ( $EnableCORS -eq "No" ) { Write-Log -LogLevel Information -Message "CORS settings not applied, only required for Development and Testing environments"
+        }
+        else {
+            Write-Log -LogLevel Information -Message "Setting Storage Context and applying CORS settings"
+            $StorageContext = New-AzureStorageContext -StorageAccountName $StorageAccountName -StorageAccountKey $SaSToken
+            Set-AzureStorageCORSRule -ServiceType Blob -CorsRules $CORSRules -Context $StorageContext
+        }
+    }
+    catch {
+        throw "Failed to get Storage Context and set CORS settings: $_"
+    }
+}
+
+function Start-ContentPurge {
+    <#
+.SYNOPSIS
+Purges the content from an Azure Content Delivery Network (CDN)
+
+.DESCRIPTION
+Purges the content from an Azure Content Delivery Network (CDN)
+
+.PARAMETER CDNProfileResourceGroup
+The Resource Group of the CDN
+
+.PARAMETER CDNProfileName
+The CDN Profile Name
+
+.PARAMETER CDNEndPointName
+The CDN EndPoint Name
+
+.PARAMETER PurgeContent
+The content to purge
```

Comment:
It's not obvious what this parameter is, can you expand on what the string should look like if it is populated?

Reply:
.PARAMETER PurgeContent
The assest you wish to purge from the edge nodes
Single URL Purge: Purge individual asset by specifying the full URL, e.g., "/pictures/image1.png" or "/pictures/image1"
Wildcard purge: Purge all folders, sub-folders, and files under an endpoint with "/*"  e.g. "/* " or "/pictures/*"
Root domain purge: Purge the root of the endpoint with "/" in the path